app/test_main.py

```diff
@@ -1 +1,99 @@
-# write your code here
+import pytest
+from unittest import mock
+from requests.exceptions import (
+    MissingSchema,
+    ConnectionError,
+    InvalidURL
+)
+
+from app.main import can_access_google_page
+
+
+@mock.patch("app.main.valid_google_url")
+def test_valid_google_url_works_in_can_access_google_page(
+        mocked_valid_url: any
+) -> None:
+    can_access_google_page("http://google.com")
+
+    mocked_valid_url.assert_called_once()
+
+
+@mock.patch("app.main.has_internet_connection")
+def test_has_internet_connection_works_in_can_access_google_page(
+        mock_internet_connection: any
+) -> None:
+    can_access_google_page("http://google.com")
+
+    mock_internet_connection.assert_called_once()
+
+
+@mock.patch("app.main.has_internet_connection")
+@pytest.mark.parametrize(
+    "initial_url,initial_hour,expected_result",
+    [
+        pytest.param(
+            "http://google.com",
+            True,
+            "Accessible",
+            id="check valid url and right hour"
```

Comment:
You need to mock the valid_google_url function in the same way as  the has_internet_connection and parametrize its return values to simulate different outcomes.